zebra-state/src/service.rs

```diff
@@ -718,98 +635,53 @@ impl StateService {
     /// Attempt to validate and commit all queued blocks whose parents have
     /// recently arrived starting from `new_parent`, in breadth-first ordering.
     #[tracing::instrument(level = "debug", skip(self, new_parent))]
-    fn process_queued(&mut self, new_parent: block::Hash) {
-        let mut new_parents: Vec<(block::Hash, Result<(), CloneError>)> =
-            vec![(new_parent, Ok(()))];
+    fn send_process_queued(&mut self, new_parent: block::Hash) {
+        use tokio::sync::mpsc::error::SendError;
+        if let Some(non_finalized_block_write_sender) = &self.non_finalized_block_write_sender {
+            let mut new_parents: Vec<block::Hash> = vec![new_parent];
+
+            while let Some(parent_hash) = new_parents.pop() {
+                let queued_children = self
+                    .queued_non_finalized_blocks
+                    .dequeue_children(parent_hash);
+
+                for queued_child in queued_children {
+                    let child_hash = queued_child.0.hash;
+
+                    let send_result = non_finalized_block_write_sender.send(
+                        NonFinalizedWriteCmd::ProcessQueued {
+                            parent_hash,
```

Comment:
Nitpick: the `parent_hash` field is redundant, because it is the same as `queued_child.0.block.header.parent_hash` (or something like that).

Reply:
I probably should've noticed this, fixed in [`a27aa3f`](https://github.com/ZcashFoundation/zebra/pull/5257/commits/a27aa3fa449dd9d9bd482f0de627f349c738d3b7)